### PR TITLE
GitHubStatusPush: Gracefully handle null 'branch'

### PR DESCRIPTION
**Pull request by zware**
## Contributor Checklist:

* [ ] I have updated the unit tests
* [x] I have created a file in the `master/buildbot/newsfragments` directory (and read the `README.txt` in that directory)
* ~~I have updated the appropriate documentation~~ No documentation update necessary.

This fixes an issue that we've had on buildbot.python.org, presumably with builders triggered by a daily scheduler rather than a push scheduler.  It seems to just leave an error message in the log, with no ill effects beyond not reporting the status.

I have not updated tests for this, as I'm not familiar enough with how the tests are set up (or how exactly GitHubStatusPush even works) to know what to update or how.  I'm happy to take suggestions, or the branch is open to you :)

### CODE REVIEW

File: master/buildbot/reporters/github.py [hunk 1]
```diff
@@ -129,12 +129,12 @@ def send(self, build):
 
         project = sourcestamps[0]['project']
 
+        issue = None
         branch = props['branch']
```
Comment (tardyp) [inline master/buildbot/reporters/github.py]: probably if there is no branch property, there should be a branch key in the sourcestamp disctionary.

Comment (p12tic) [reply master/buildbot/reporters/github.py]: Indeed, we should check the sourcestamp for the branch. It's not guaranteed to be there though.

**Comment by p12tic** [comment]
The tests live in test_reporter_github.py file, insertTestData in test/util/reporter.py will need to be updated with a new argument to not insert branch property. Long term we'll need to refactor that function to allow full customization of test data, but for now let's just do the minimum.

File: master/buildbot/newsfragments/fix-githubstatuspush-null-branch.bugfix [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Fixed handling
```
Comment (sa2ajj) [inline master/buildbot/newsfragments/fix-githubstatuspush-null-branch.bugfix]: should this be a single line?

**Comment by p12tic** [comment]
Removing `stalled` label as the code has ended up in the codebase by some other PR.
